lua/lean/infoview.lua

```diff
@@ -131,21 +146,72 @@ function M.open()
   return M._infoviews
 end
 
-function M.close()
-  if not M.is_open() then return end
-  local src_win = vim.api.nvim_get_current_win()
+function M.set_pertab()
+  if vim.g.lean_info_pertab then return end
+
+  M.close_all_info()
+
+  vim.g.lean_info_pertab = true
+
+  M.set_update()
+end
+
+function M.set_perwindow()
+  if not vim.g.lean_info_pertab then return end
+
+  M.close_all_info()
+
+  vim.g.lean_info_pertab = false
+
+  M.set_update()
+end
+
+function M.close_all_info()
+  -- close all current infoviews
+  for key, _ in pairs(M._infoviews) do
+    M.close_win(key)
+  end
+  for key, _ in pairs(M._infoviews_open) do
+    M._infoviews_open[key] = nil
+  end
+end
 
-  if M._infoviews[src_win].win then
-    vim.api.nvim_win_close(M._infoviews[src_win].win, true)
+function M.close_win(src_winnr)
+  if M._infoviews[src_winnr] then
+    vim.api.nvim_win_close(M._infoviews[src_winnr].win, true)
   end
 
   -- NOTE: it seems this isn't necessary since unlisted buffers are deleted automatically?
```

Comment:
I added this I believe while seeing tests fail for different reasons and causing later tests to fail. But it's possible it indeed isn't necessary. If it doesn't seem to be, feel free to kill it.